diesel/src/expression/helper_types.rs

```diff
@@ -87,13 +87,39 @@ pub type Asc<Expr> = super::operators::Asc<Expr>;
 /// [`expr.nullable()`](../expression_methods/trait.NullableExpressionMethods.html#method.nullable)
 pub type Nullable<Expr> = super::nullable::Nullable<Expr>;
 
+/// The return type of
+/// [`lhs.and(rhs)`](../expression_methods/trait.BoolExpressionMethods.html#method.and)  
+/// when resulting expression is a non-nullable `Bool`
+pub type And<Lhs, Rhs> = MaybeNullableAnd<Lhs, Rhs, sql_types::Bool>;
```

Comment:
Instead of having multiple variants here I would like to have just one:

```suggestion
pub type And<Lhs, Rhs, ST = sql_types::Bool> = Grouped<super::operators::And<Lhs, AsExprOf<Rhs, ST>>>;
```